packages/mongodb-ts-autocomplete/scripts/extract-types.ts

```diff
@@ -1,43 +1,257 @@
+    'http.d.ts',
+    'http2.d.ts',
+    'https.d.ts',
+    'index.d.ts',
+    'inspector.d.ts',
+    'module.d.ts',
+    'net.d.ts',
+    'os.d.ts',
+    'path.d.ts',
+    'perf_hooks.d.ts',
+    'process.d.ts',
+    'punycode.d.ts',
+    'querystring.d.ts',
+    'readline.d.ts',
+    'readline/promises.d.ts',
+    'repl.d.ts',
+    'sea.d.ts',
+    'sqlite.d.ts',
+    'stream.d.ts',
+    'stream/consumers.d.ts',
+    'stream/promises.d.ts',
+    'stream/web.d.ts',
+    'string_decoder.d.ts',
+    'test.d.ts',
+    'timers.d.ts',
+    'timers/promises.d.ts',
+    'tls.d.ts',
+    'trace_events.d.ts',
+    'tty.d.ts',
+    'url.d.ts',
+    'util.d.ts',
+    'v8.d.ts',
+    'vm.d.ts',
+    'wasi.d.ts',
+    'worker_threads.d.ts',
+    'zlib.d.ts',
+  ],
+  assert: [
+    'package.json',
+    'assert.js', // exists only
+  ],
+  buffer: ['package.json', 'index.d.ts'],
+  events: ['package.json'],
+  punycode: [
+    'package.json',
+    'punycode.js', // exists only
+  ],
+  querystring: [
+    'package.json',
+    'index.js', // exists only
+  ],
+  string_decoder: [
+    'package.json',
+    'lib/string_decoder.js', // exists only
+  ],
+  typescript: [
+    'package.json',
+    'lib/es2023.ts',
+    'lib/lib.decorators.d.ts',
+    'lib/lib.decorators.legacy.d.ts',
+    'lib/lib.es2015.collection.d.ts',
+    'lib/lib.es2015.core.d.ts',
+    'lib/lib.es2015.d.ts',
+    'lib/lib.es2015.generator.d.ts',
+    'lib/lib.es2015.iterable.d.ts',
+    'lib/lib.es2015.promise.d.ts',
+    'lib/lib.es2015.proxy.d.ts',
+    'lib/lib.es2015.reflect.d.ts',
+    'lib/lib.es2015.symbol.d.ts',
+    'lib/lib.es2015.symbol.wellknown.d.ts',
+    'lib/lib.es2016.array.include.d.ts',
+    'lib/lib.es2016.d.ts',
+    'lib/lib.es2016.intl.d.ts',
+    'lib/lib.es2017.arraybuffer.d.ts',
+    'lib/lib.es2017.d.ts',
+    'lib/lib.es2017.date.d.ts',
+    'lib/lib.es2017.intl.d.ts',
+    'lib/lib.es2017.object.d.ts',
+    'lib/lib.es2017.sharedmemory.d.ts',
+    'lib/lib.es2017.string.d.ts',
+    'lib/lib.es2017.typedarrays.d.ts',
+    'lib/lib.es2018.asyncgenerator.d.ts',
+    'lib/lib.es2018.asynciterable.d.ts',
+    'lib/lib.es2018.d.ts',
+    'lib/lib.es2018.intl.d.ts',
+    'lib/lib.es2018.promise.d.ts',
+    'lib/lib.es2018.regexp.d.ts',
+    'lib/lib.es2019.array.d.ts',
+    'lib/lib.es2019.d.ts',
+    'lib/lib.es2019.intl.d.ts',
+    'lib/lib.es2019.object.d.ts',
+    'lib/lib.es2019.string.d.ts',
+    'lib/lib.es2019.symbol.d.ts',
+    'lib/lib.es2020.bigint.d.ts',
+    'lib/lib.es2020.d.ts',
+    'lib/lib.es2020.date.d.ts',
+    'lib/lib.es2020.intl.d.ts',
+    'lib/lib.es2020.number.d.ts',
+    'lib/lib.es2020.promise.d.ts',
+    'lib/lib.es2020.sharedmemory.d.ts',
+    'lib/lib.es2020.string.d.ts',
+    'lib/lib.es2020.symbol.wellknown.d.ts',
+    'lib/lib.es5.d.ts',
+  ],
+  'undici-types': ['package.json', 'index.d.ts'],
+  url: [
+    'package.json',
+    'url.js', // exists only
+  ],
+  util: [
+    'package.json',
+    'util.js', // exists only
+  ],
```

Comment:
I'm sure you've had a similar thought, but is there something we can do to make this list a bit easier to maintain (i.e. auto-generate it)?

Reply:
Best idea I've had so far (and I thought about this for a couple of days, trying different things) is the fallback servicesHost that gathers the ones we're missing and fails a test. That's exactly how I build this list.

I don't expect this to break at all unless we bump typescript or @types/node and even then probably rarely. Personally I think we can see how it goes and optimise that at a later stage if it becomes a problem.

Whatever we do I'd like to keep a human in the loop because automatically shipping everything typescript decides to touch during module resolution is probably at least a little bit dangerous.

Reply:
But I'm definitely open to suggestions and happy to build whatever people come up with!